### ISSUE TITLE
PDF title metadata contains ` -- ` even when `\title{}` is never called

### ISSUE THREAD

**Issue by devurandom** [moderncv/moderncv#110]
When generating a PDF without setting `\title{...}` but only setting `\name{Max}{Mustermann}`, the title metadata of the PDF is set to `Max Mustermann – `, even though `/usr/share/texmf-dist/tex/latex/moderncv/moderncv.cls` uses `\hypersetup{pdftitle = {\@firstname{}~\@lastname{}\notblank{\@title}{ -- \@title}{}}}` and I would thus expect the title to be `Max Mustermann` only.

Minimal example:

```latex
\documentclass{moderncv}

\name{Max}{Mustermann}

\begin{document}
Text
\end{document}
```

```
❯ pdfinfo resume.pdf | grep -E '^Title:'
Title:           Max Mustermann – 
```

I use latexmk 4.70b and moderncv 2.1.0 from TeXLive 2021 to produce the PDF with following `latexmkrc`:

```perl
$pdf_mode = 4;
```

```
❯ grep Provides /usr/share/texmf-dist/tex/latex/moderncv/moderncv.cls
\ProvidesClass{moderncv}[2021/01/21 v2.1.0 modern curriculum vitae and letter document class]

❯ latexmk --version

Latexmk, John Collins, 29 September 2020. Version 4.70b
```

**Comment by mefuller** [comment]
I'm unable to reproduce this.

Would you please verify that you are using the latest version and submit a minimum working example?

**Comment by devurandom** [comment]
I rebuilt using v2.3.1 checked out from Git and was able to reproduce the behaviour using the minimal working example I provided above ("minimal example"):

```latex
\documentclass{moderncv}

\name{Max}{Mustermann}

\begin{document}
Text
\end{document}
```

To use a local copy of moderncv instead of the system-provided one, I changed `latexmkrc` to:

```perl
$ENV{TEXINPUTS} = ".:./latex/moderncv:" . ($ENV{TEXINPUTS} or "");

$pdf_mode = 4;
```

First lines of the relevant output are:

```
This is LuaHBTeX, Version 1.13.0 (TeX Live 2021 Gentoo Linux)
 restricted system commands enabled.
(./main.tex
LaTeX2e <2020-10-01> patch level 4
 L3 programming layer <2021-02-18> (./latex/moderncv/moderncv.cls
Document Class: moderncv 2022-02-21 v2.3.1 modern curriculum vitae and letter d
ocument class
```

**Comment by mefuller** [comment]
I see now that you mean the PDF metadata has the title containing `--`, not that `--` is being printed on the page.
This was not at all clear previously.

I'll have to dig around to determine why this occurs later.

**Comment by tianyizheng02** [comment]
@mefuller This issue should be closed since #116 fixes it

**Comment by mefuller** [comment]
Closed by #116
